### PR TITLE
KIN-264 create certificate table

### PR DESCRIPTION
**Pull request by FernandoJoachin**
[KIN-264](https://loopcrack.atlassian.net/browse/KIN-264) Medium [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=loopcrack-org&utm_campaign=badge_info)

### Problem

As an admin panel user, I want in the admin panel to see all the certificate records, edit those records, delete certificates and create new ones so that I can manage the certificates that will be displayed in the public view. .

### Solution

Make a page where you can view all certificate records, edit those records, delete certificates, and create new ones so you can manage the certificates that will be displayed in public view. .

### Requirements checklist

- [x] It must display the certificates records on a table, where it would display the following info: Certificate Name, Certificate image preview and Edit and delete buttons on the same column.
- [x] It must have a search bar that filters by name.
- [x] It must have a create button above the table of the records

### Testing

- [ ] Go to the next route: http://kinub.cm/admin/certificados

## Demo

[](https://postimg.cc/hX1RQVfw)

[KIN-264]: https://loopcrack.atlassian.net/browse/KIN-264?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/Views/admin/certificates/Certificates.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                    <?php
+                        if (session()->has('response')) {
+                            $response = session()->get('response');
+                            ?>
+                        <div id="alertElement" data-response="<?= htmlspecialchars(json_encode($response)) ?>"></div>
+                    <?php } ?>
+
+                    <!-- card-header -->
+                    <div class="card-body">
+                        <table id="certificates-table" class="table nowrap dt-responsive align-middle table-hover table-bordered" style="width:100%">
```
Comment (Oswaldo-Chan) [inline app/Views/admin/certificates/Certificates.php:41]: Please remove the style line and replace it by adding the `w-100` class

File: app/Views/admin/certificates/Certificates.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                                            </div>
+                                        </td>
+                                        <td>
+                                            <div class="d-flex justify-content-center align-items-center">
+                                                <img src="<?= $certificate['fileRoute']; ?>" alt="Imagen del certificado " . <?= $certificate['certificateId']; ?> width="200" height="100">
+                                            </div>
+                                        </td>
+                                        <td>
+                                            <div class="d-flex flex-column justify-content-between">
+                                                <a href="/admin/certificados/editar/<?= $certificate['certificateId']; ?>" class="btn btn-primary btn-icon waves-effect waves-light mb-2" style="width: 100%"><i class="ri-edit-2-fill ri-lg"></i></a>
```
Comment (Oswaldo-Chan) [inline app/Views/admin/certificates/Certificates.php:66]: Please remove the style line and replace it by adding the `w-100` class

File: app/Views/admin/certificates/Certificates.php [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                                                <a href="#" class="btn btn-danger btn-icon waves-effect waves-light" data-bs-toggle="modal" style="width: 100%;"><i class="ri-delete-bin-5-line ri-lg"></i></a>
```
Comment (Oswaldo-Chan) [inline app/Views/admin/certificates/Certificates.php:67]: Please remove the style line and replace it by adding the `w-100` class

**Review by Oswaldo-Chan — CHANGES_REQUESTED** [review]
I was reviewing the files with @Cristianpan and we noticed that this view implements styles in the HTML when it can use bootstrap classes.

File: app/Views/admin/certificates/Certificates.php [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                                    <tr>
+                                        <td><?= $certificate['certificateId']; ?></td>
+                                        <td>
+                                            <div class="d-flex align-items-center fw-medium">
+                                                <p class="text-wrap"><?= $certificate['certificatefileName'] ?></p>
+                                            </div>
+                                        </td>
+                                        <td>
+                                            <div class="d-flex justify-content-center align-items-center">
+                                                <img src="<?= $certificate['fileRoute']; ?>" alt="Imagen del certificado " . <?= $certificate['certificateId']; ?> width="200" height="100">
```
Comment (Cristianpan) [inline app/Views/admin/certificates/Certificates.php:61]: Add the "mw-100h-auto" classes
```suggestion
                                                <img class="mw-100 h-auto" src="<?= $certificate['fileRoute']; ?>" alt="Imagen del certificado " . <?= $certificate['certificateId']; ?> width="200" height="100">
```

**Review by Cristianpan — CHANGES_REQUESTED** [review]
Good job @FernandoJoachin , however I have noticed along with @Oswaldo-Chan some details in the design, mainly in the galaxy fold view. 
As you can see, when you want to see more details of the certificate, the table overflows.
image

Likewise, it's something very subtle but the aspect ratio of the image doesn't correspond to the original proportion, so it looks deformed. 
image

I have left the suggestion so that you can correct those details.

**Review by LUGAMAFE — APPROVED** [review]
Replace all `style="width:100%"`  with `w-100` as @Oswaldo-Chan mention 

And don´t worry about this overflow its a problem with datatables  image

**Review by Mauriciocr207 — APPROVED** [review]

**Comment by FernandoJoachin** [comment]
> Nice work!, we just saw a little problem with the certificate images. At resolutions 600px wide or smaller the images start to deform. We thought you could activate the mobile view at 600px or make the image resize when reaching 600px.
> 
> CC: @CheloChidoone

I already solved the problem. Add the following to the image: object-fit-contain

**Comment by Mauriciocr207** [comment]
> > Nice work!, we just saw a little problem with the certificate images. At resolutions 600px wide or smaller the images start to deform. We thought you could activate the mobile view at 600px or make the image resize when reaching 600px.
> > CC: @CheloChidoone
> 
> I already solved the problem. Add the following to the image: object-fit-contain CC: @Mauriciocr207

Wee see that if you set the image with the property :
```css
object-fit: content
```
the image mantain itsaspect

**Review by LUGAMAFE — APPROVED** [review]

**Review by CarlosMay7 — APPROVED** [review]

**Review by Oswaldo-Chan — APPROVED** [review]

**Review by Cristianpan — APPROVED** [review]

**Review by CarlosJCD — APPROVED** [review]

**Pull request merged**